### PR TITLE
cross-platform shell for npm run scripts

### PR DESCRIPTION
**Pull request by gucong3000**
Closes: #18580

### CODE REVIEW

File: node_modules/npm-lifecycle/index.js [hunk 1]
```diff
@@ -262,7 +262,7 @@ function runCmd_ (cmd, pkg, env, wd, opts, stage, unsafe, uid, gid, cb_) {
 
   if (customShell) {
     sh = customShell
-  } else if (process.platform === 'win32') {
```
Comment (iarna) [inline node_modules/npm-lifecycle/index.js:265]: This file is in a dependency. To patch this you need to PR this change to [npm-lifecycle](https://github.com/npm/npm-lifecycle).

I think this is the right approach though.  When you PR this to `npm-lifecycle` make sure you include a default value for the `isWindowsShell` option that's compatible.

**Review by iarna — CHANGES_REQUESTED** [review]

File: node_modules/npm-lifecycle/index.js (hunk 1 above)
Comment (gucong3000) [reply node_modules/npm-lifecycle/index.js:265]: done.

**Comment by gucong3000** [comment]
@iarna Is there anything else I need to do?

**Comment by zkat** [comment]
Hi! We're moving repos to https://github.com/npm/cli/pulls! See [our blog post about the migration to npm.community](https://blog.npmjs.org/post/175587538995/announcing-npmcommunity) for details. As such, we're closing all active PRs on this repo.

Could you please re-open this PR against `npm/cli` instead? We're still interested in this patch so we hope you will! We'll continue discussion over there once it's moved. 💚

**Comment by zkat** [comment]
p.s. You need to update `npm-lifecycle` in its own repo and not include the patch directly like this.

**Pull request closed without merging**